inference-engine/src/mkldnn_plugin/mkldnn_graph_optimizer.cpp

```diff
@@ -1812,3 +1817,57 @@ void MKLDNNGraphOptimizer::MergeTransposeAndReorder(MKLDNNGraph &graph) {
         }
     }
 }
+
+void MKLDNNGraphOptimizer::reshapeRnnSeq(MKLDNNGraph &graph) {
+    auto& graphNodes = graph.GetNodes();
+
+    auto isSutableParentNode = [](MKLDNNNodePtr node) {
+        auto rnnNode = std::dynamic_pointer_cast<MKLDNNRNN>(node);
+        return rnnNode && !rnnNode->hasNativeOrder() && node->outDims[0].ndims() == 4 && node->outDims[0][1] == 1;
+    };
+
+    auto isSutableChildNode = [&](MKLDNNNodePtr parentNode, MKLDNNNodePtr childNode) {
+        return childNode->inDims[0].ndims() == 4 && childNode->inDims[0][1] == 1;
+    };
+
+    auto parent = graphNodes.begin();
+    while (parent != graphNodes.end()) {
+        auto parentNode = *parent;
+        if (!isSutableParentNode(parentNode)) {
+            parent++;
+            continue;
+        }
+
+        auto childNode = parentNode->getChildEdgeAt(0)->getChild();
+        if (!isSutableChildNode(parentNode, childNode)) {
+            parent++;
+            continue;
+        }
+
+        auto parents = childNode->parentEdges;
+
+        for (size_t i = 0; i < parents.size(); i++) {
+            auto p_edge = parents[i].lock();
+            if (!p_edge) continue;
+            auto parent = p_edge->getParent();
+            if (!parent) continue;
+
+            if (parent == parentNode) {
+                auto newRnnOutDims = parentNode->outDims[0].ToSizeVector();
+                newRnnOutDims.erase(newRnnOutDims.begin() + 1);
+                parentNode->outDims[0] = MKLDNNDims{newRnnOutDims};
+
+                const MKLDNNNodePtr newReshape = std::make_shared<MKLDNNReshapeNode>(parent->getName() + "_abc_a1bc", graph.getEngine(), graph.weightsCache);
+                newReshape->inDims.push_back(parentNode->outDims[0]);
```

Comment:
can we fill inDims/outDims and precision into ctor?

Reply:
Fixed: parent node is passed into ctor and dims with precisions are filled there. Child node was not due to in general it may not exist when this ctor is called.